src/torchcodec/_core/BetaCudaDeviceInterface.cpp

```diff
@@ -0,0 +1,563 @@
+// Copyright (c) Meta Platforms, Inc. and affiliates.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <torch/types.h>
+#include <mutex>
+#include <vector>
+
+#include "src/torchcodec/_core/BetaCudaDeviceInterface.h"
+
+#include "src/torchcodec/_core/DeviceInterface.h"
+#include "src/torchcodec/_core/FFMPEGCommon.h"
+#include "src/torchcodec/_core/NVDECCache.h"
+
+#include <cuda_runtime.h> // For cudaStreamSynchronize
```

Comment:
I don't see `cudaStreamSynchronize` referenced anywhere - is the include still needed?

Reply:
Good catch thank you - that a remnant from a previous implem. I'll remove (although we'll very likely need it back at some point!)